internal/sequence/sequence.go

```diff
@@ -0,0 +1,131 @@
+package sequence
+
+import (
+	"encoding/binary"
+
+	"cosmossdk.io/errors"
+	"cosmossdk.io/math"
+	storetypes "github.com/cosmos/cosmos-sdk/store/types"
+)
+
+var (
+	ErrSequenceUniqueConstraint = errors.Register("sequence_u256", 1, "sequence already initialized")
+)
+
+type Number interface {
+	uint32 | math.Uint
+}
+
+type Sequence[T Number] struct {
+	storeKey []byte
+}
+
+func NewSequence[T Number](prefix []byte) Sequence[T] {
+	return Sequence[T]{
+		storeKey: prefix,
+	}
+}
+
+func (s Sequence[T]) NextVal(store storetypes.KVStore) T {
+	v := store.Get(s.storeKey)
+	seq := s.DecodeSequence(v)
+	seq = s.IncreaseSequence(seq)
+	store.Set(s.storeKey, s.EncodeSequence(seq))
+	return any(seq).(T)
+}
+
+// CurVal returns the last value used. 0 if none.
+func (s Sequence[T]) CurVal(store storetypes.KVStore) T {
+	v := store.Get(s.storeKey)
+	ret := s.DecodeSequence(v)
+	return any(ret).(T)
+}
+
+// PeekNextVal returns the CurVal + increment step. Not persistent.
+func (s Sequence[T]) PeekNextVal(store storetypes.KVStore) T {
```

Comment:
Would suggest we delete `PeekNextVal` if it is not used now.

Reply:
OK，will delete it.